### DISCUSSION TITLE
I need to connect to chatroom with password

### DISCUSSION THREAD

**Discussion by PepeMax** [xmppjs/xmpp.js#960] [Q&A] [answered]
Hello for my project I need to connect to a chatroom protected by password how can I do?

Thanks

**Comment by sonnyp** [comment]
https://xmpp.org/extensions/xep-0045.html#enter-pw

**Reply by PepeMax** [reply]
Thanks,

I have already read this passage but how do I integrate it into my code ?

How do I add this : 
```
<x xmlns='http://jabber.org/protocol/muc'>
    <password>cauldronburn</password>
</x>
```

**Reply by sonnyp** [reply]
https://github.com/xmppjs/xmpp.js/blob/main/packages/client/example.js#L32

See https://github.com/xmppjs/xmpp.js/tree/main/packages/xml

**Comment by PepeMax** [comment, marked as answer]
I didn't understand how it worked but now it's good !

For example :
`let message= xml('presence', { to: room_jid }, xml('x' ,{xmlns : 'http://jabber.org/protocol/muc'}, xml('password', {}, 'test')))
`
Thank you
